test/extended/networking/internal_ports.go

```diff
@@ -0,0 +1,80 @@
+package networking
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2enetwork "k8s.io/kubernetes/test/e2e/framework/network"
+
+	"github.com/onsi/ginkgo"
+)
+
+const (
+	nodeTCPPort = 9000
+	nodeUDPPort = 9999
+)
+
+func newNetworkingTestConfig(f *framework.Framework) *e2enetwork.NetworkingTestConfig {
+	return e2enetwork.NewNetworkingTestFromConfig(f, &e2enetwork.NetworkingTestConfig{
+		NodeHTTPPort: nodeTCPPort,
+		NodeUDPPort:  nodeUDPPort,
+	})
+}
+
+var _ = ginkgo.Describe("[sig-network][Feature:Platform] Internal connectivity", func() {
+	f := framework.NewDefaultFramework("nettest")
+
+	ginkgo.It("should function for pod-Service: http", func() {
+		config := newNetworkingTestConfig(f)
+		ginkgo.By(fmt.Sprintf("dialing(http) %v --> %v:%v (config.clusterIP)", config.TestContainerPod.Name, config.ClusterIP, nodeTCPPort))
+		config.DialFromTestContainer("http", config.ClusterIP, nodeTCPPort, config.MaxTries, 0, config.EndpointHostnames())
```

Comment:
I looked through `vendor/k8s.io/kubernetes/test/e2e/framework/networking_utils.go` briefly to try and understand `EndpointHostnames`, but it's like an onion.  Can anyone give a high-level summary about whether we can expect these to run on compute and/or control-plane machines?

Reply:
They are randomly chosen.

Reply:
In this PR this is validated by tolerating master taints.